### PR TITLE
Use zone to filter PowerVS service instance

### LINKED ISSUE #1839: Need a check on powervs service instance with zone since dup name is allowed
/kind bug
/area provider/ibmcloud

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
Since powervs service allows duplicate names when I try to create powervs instance on a specific region, if there is powervs instance already exists with the same name in different region, currently capi takes it and assigns it instead of creating a new one in given region.  

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by dharaneeshvrd**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1839 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Review by Karthik-K-N — COMMENTED** [review]
Changes looks good to me, But trying to understand when there were two service instance with same name , why didn't you hit this error [here](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1853/files#diff-c1cb986cbf553e2fe5a2a8955848ea3044570c1d27c74efef8553f3c8289856cR143)
```
default:
		errStr := fmt.Errorf("there exist more than one service instance ID with same name %s, Try setting serviceInstance.ID", name)
```

Also within the same zone we can't have the same service instance right?

**Comment by dharaneeshvrd** [comment]
Problem I faced is, let's say there is an existing powervs instance called `capi-dhar-serviceInstance` in osa21 zone.
When I try to deploy new cluster with same name `capi-dhar` in dal12 zone, instead of creating a new powervs instance in dal12, it tries to use the one in osa21 since the it's having same name.
It won't hit [this](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1853/files#diff-c1cb986cbf553e2fe5a2a8955848ea3044570c1d27c74efef8553f3c8289856cR143), since there no two matching instances in this situation.

> Also within the same zone we can't have the same service instance right? 
But for this unfortunately it still allows duplicate within zone too.

**Comment by Karthik-K-N** [comment]
> Problem I faced is, let's say there is an existing powervs instance called `capi-dhar-serviceInstance` in osa21 zone. When I try to deploy new cluster with same name `capi-dhar` in dal12 zone, instead of creating a new powervs instance in dal12, it tries to use the one in osa21 since the it's having same name. It won't hit [this](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1853/files#diff-c1cb986cbf553e2fe5a2a8955848ea3044570c1d27c74efef8553f3c8289856cR143), since there no two matching instances in this situation.
> 
> > Also within the same zone we can't have the same service instance right?
> 
> But for this unfortunately it still allows duplicate within zone too.

Got the point now, Thank you.

**Review by Karthik-K-N — COMMENTED** [review]
/lgtm

**Comment by mkumatag** [comment]
> But for this unfortunately it still allows duplicate within zone too.

This is exactly my doubt too, we can't foolproof 100% here in this case, we need to mention somewhere in the api spec saying that if multiple instance with same name present then it will pick first in the list and if user wants to use any specific one then use by ID

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: cloud/scope/powervs_machine.go [hunk 1]
```diff
@@ -178,9 +179,10 @@ func NewPowerVSMachineScope(params PowerVSMachineScopeParams) (scope *PowerVSMac
 		serviceInstanceName = fmt.Sprintf("%s-%s", params.IBMPowerVSCluster.GetName(), "serviceInstance")
 		if params.IBMPowerVSCluster.Spec.ServiceInstance != nil && params.IBMPowerVSCluster.Spec.ServiceInstance.Name != nil {
 			serviceInstanceName = *params.IBMPowerVSCluster.Spec.ServiceInstance.Name
+			zone = params.IBMPowerVSCluster.Spec.Zone
```
Comment (mkumatag) [inline cloud/scope/powervs_machine.go:182]: will it make any difference? why don't we just pass params.IBMPowerVSCluster.Spec.Zone directly in the below function?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: cloud/scope/powervs_machine.go (hunk 1 above)
Comment (dharaneeshvrd) [reply cloud/scope/powervs_machine.go]: Agree. changed.

**Comment by dharaneeshvrd** [comment]
Updated the ServiceInstance field's description, ptal!

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: api/v1beta2/ibmpowervscluster_types.go [hunk 2]
```diff
@@ -65,6 +65,7 @@ type IBMPowerVSClusterSpec struct {
 	// when omitted system will dynamically create the service instance with name CLUSTER_NAME-serviceInstance.
 	// when ServiceInstance.ID is set, its expected that there exist a service instance in PowerVS workspace with id or else system will give error.
 	// when ServiceInstance.Name is set, system will first check for service instance with Name in PowerVS workspace, if not exist system will create new instance.
+	// if there are more than one service instance exist with the ServiceInstance.Name in given Zone, installation fails with an error. Use ServiceInstance.ID in those situations to use the specific service instance.
```
[the remark this answers is not in the document]
Comment (dharaneeshvrd) [reply api/v1beta2/ibmpowervscluster_types.go:68]: @mkumatag Here https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1853/files#diff-c1cb986cbf553e2fe5a2a8955848ea3044570c1d27c74efef8553f3c8289856cR143

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**